src/table.scss

```diff
@@ -3,355 +3,415 @@
+    .#{$block}__row {
+      border-bottom: $fd-table-border;
 
-    @include fd-hover() {
-      @include fd-var-color("background-color", $fd-table-row-background-color--hover, --fd-color-background-hover);
+      &:hover {
+        background-color: inherit;
+      }
     }
 
-    @include fd-selected() {
-      @include fd-var-color("background-color", $fd-table-row-background-color--selected, --fd-color-background-selected);
+    .#{$block}__cell {
+      height: $fd-table-cozy-cell-height;
+      background-color: var(--sapList_HeaderBackground);
+      color: var(--sapList_HeaderTextColor);
 
       @include fd-hover() {
-        @include fd-var-color("background-color", fd-color-state("selected-hover"), --fd-color-background-selected-hover);
+        background-color: var(--sapList_Hover_Background);
       }
+
+      @include fd-trigger-element();
     }
+  }
 
-    &--success,
-    &--warning,
-    &--error,
-    &--information {
-      border: none;
+  &__cell {
+    text-align: left;
+    height: $fd-table-cozy-cell-height;
+    padding: 0 $fd-table-cell-padding;
+    color: inherit;
+    border-right: $fd-table-border;
+    word-break: break-word;
+
+    &--valid {
+      color: var(--sapSuccessColor);
+    }
 
-      .#{$block}__cell:first-child {
-        position: relative;
-
-        &::before {
-          position: absolute;
-          left: 0;
-          top: 0;
-          border: fd-space(0.5) solid;
-          border-top-left-radius: fd-space(1);
-          border-bottom-left-radius: fd-space(1);
-          content: "";
-          height: 100%;
-        }
+    &--warning {
+      color: var(--sapWarningColor);
+    }
 
-        @include fd-rtl() {
-          &::before {
-            left: auto;
-            right: 0;
-            border-top-left-radius: 0;
-            border-bottom-left-radius: 0;
-            border-top-right-radius: fd-space(1);
-            border-bottom-right-radius: fd-space(1);
-          }
-        }
-      }
+    &--error {
+      color: var(--sapErrorColor);
     }
 
-    &--success {
-      @include fd-var-color("background-color", fd-color('background', 3), --fd-color-background-3);
+    &--information {
+      color: var(--sapInformationColor);
+    }
 
-      .#{$block}__cell:first-child {
-        &::before {
-          @include fd-var-color("border-color", fd-color('status', 1), --fd-color-status-1);
-        }
-      }
+    &--checkbox {
+      width: 10px;
+      padding: 0;
+      margin: 0;
     }
 
-    &--warning {
-      @include fd-var-color("background-color", fd-color('background', 4), --fd-color-background-4);
+    &--fit-content {
+      width: 10px;
+      word-break: keep-all;
+      white-space: nowrap;
+    }
 
-      .#{$block}__cell:first-child {
-        &::before {
-          @include fd-var-color("border-color", fd-color('status', 2), --fd-color-status-2);
-        }
-      }
+    &--no-padding {
+      padding: 0;
+      margin: 0;
```

Comment:
since we see this twice in the component you can create a local reset mixin and reuse it